sim/team-validator.js

```diff
@@ -921,6 +921,22 @@ class Validator {
 		while (template && template.species && !alreadyChecked[template.speciesid]) {
 			alreadyChecked[template.speciesid] = true;
 			if (dex.gen === 2 && template.gen === 1) tradebackEligible = true;
+			// STABmons Move Legality (for tournaments)
+			// @ts-ignore
+			let restrictedMoves = format.restrictedMoves || [];
+			if (ruleTable.has('stabmonsmovelegality') && !restrictedMoves.includes(move.name) && !move.isZ) {
+				let types = template.types;
+				let baseTemplate = dex.getTemplate(template.baseSpecies);
+				for (let i in baseTemplate.otherFormes) {
+					let forme = dex.getTemplate(baseTemplate.otherFormes[i]);
+					if (baseTemplate.otherFormes && !forme.battleOnly) {
+						if (forme.forme !== 'Alola' && forme.forme !== 'Alola-Totem') {
+							types = types.concat(forme.types).concat(baseTemplate.types);
+						}
+					}
+				}
+				if (types.includes(move.type)) return false;
```

Comment:
In case this was remotely unclear:

`checkLearnset` is a function that returns `null` if the learnset is valid, and an object containing a learnset error, if the learnset is invalid.

Returning `false` is thus obviously a bug, because it is neither `null` nor an object. This is why TypeScript is yelling at you.